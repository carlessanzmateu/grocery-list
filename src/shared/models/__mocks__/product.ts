import Product from '../product';

export const ProductMock = new Product(
  'someId',
  'someImageUrl',
  10,
  'someProductName',
  100,
  'someProductDescription',
  '0',
  1
);
