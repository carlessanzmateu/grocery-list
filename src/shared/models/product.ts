export default class Product {
  private $id: string;
  private $imageUrl: string;
  private $stock: number;
  private $productName: string;
  private $price: number;
  private $productDescription: string;
  private $favorite: string;
  private $quantitySelected: number;

  constructor(
    id: string,
    imageUrl: string,
    stock: number,
    productName: string,
    price: number,
    productDescription: string,
    favorite: string,
    quantitySelected: number
  ) {
    this.$id = id;
    this.$imageUrl = imageUrl;
    this.$stock = stock;
    this.$productName = productName;
    this.$price = price;
    this.$productDescription = productDescription;
    this.$favorite = favorite;
    this.$quantitySelected = quantitySelected;
  }

  getId(): string {
    return this.$id;
  }

  getImageUrl(): string {
    return this.$imageUrl;
  }

  getStock(): number {
    return this.$stock;
  }

  getProductName(): string {
    return this.$productName;
  }

  getPrice(): number {
    return this.$price;
  }

  getProductDescription(): string {
    return this.$productDescription;
  }

  getFavorite(): string {
    return this.$favorite;
  }

  setFavorite(favorite: string) {
    this.$favorite = favorite;
  }

  getQuantitySelected(): number {
    return this.$quantitySelected;
  }

  setQuantitySelected(quantity: number) {
    this.$quantitySelected = quantity;
  }
}