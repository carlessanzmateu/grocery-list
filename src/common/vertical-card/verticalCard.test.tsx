import React from 'react';
// import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import VerticalCard from './verticalCard';

import { ProductMock } from '../../shared/models/__mocks__/product';

describe('VerticalCard', () => {
  test('should create', () => {
    function onItemSelected() {}
    function onItemAddedToFavorite() {}
  
    const component = renderer.create(
      <VerticalCard
        key={ProductMock.getId()}
        item={ProductMock}
        onItemSelected={onItemSelected}
        onItemAddedToFavorite={onItemAddedToFavorite}
      />
    );
    expect(component).toBeTruthy();
  });

  test('should match snapshot', () => {
    function onItemSelected() {}
    function onItemAddedToFavorite() {}
  
    const component = renderer.create(
      <VerticalCard
        key={ProductMock.getId()}
        item={ProductMock}
        onItemSelected={onItemSelected}
        onItemAddedToFavorite={onItemAddedToFavorite}
      />
    );

    let tree = component.toJSON();
    console.log(component)
    expect(tree).toMatchSnapshot();
  })

  // test('should have getFavoriteCssClasses method', () => {
  //   function onItemSelected() {}
  //   function onItemAddedToFavorite() {}
  
  //   const componentRender = renderer.create(
  //     <VerticalCard
  //       key={ProductMock.getId()}
  //       item={ProductMock}
  //       onItemSelected={onItemSelected}
  //       onItemAddedToFavorite={onItemAddedToFavorite}
  //     />
  //   );

  //   console.log(componentRender);
  //   expect(VerticalCard.getFavoriteCssClasses).toBeDefined();
  // })
});
