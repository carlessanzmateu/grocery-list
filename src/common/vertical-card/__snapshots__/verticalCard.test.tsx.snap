// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalCard should match snapshot 1`] = `
<div
  className="vertical-card"
>
  <span
    className="favorite"
    onClick={[Function]}
  />
  <img
    alt="someImageUrl"
    className="image"
    src="someImageUrl"
  />
  <div
    className="card-information"
  >
    <h3
      className="title"
    >
      <strong>
        someProductName
      </strong>
    </h3>
    <p
      className="description"
    >
      someProductDescription
    </p>
    <div
      className="buy-info"
    >
      <span
        className="products-left"
      >
        10
         left
      </span>
      <a
        className="add-btn"
        onClick={[Function]}
      >
        + add
      </a>
    </div>
  </div>
</div>
`;
